protobuf/acm.proto

```diff
@@ -23,4 +23,5 @@ message Account {
     uint64 Balance = 4;
     bytes Code = 5 [(gogoproto.customtype) = "Bytecode", (gogoproto.nullable) = false];
     permission.AccountPermissions Permissions = 6 [(gogoproto.nullable) = false];
+    bytes WASM = 7 [(gogoproto.customtype) = "Bytecode"];
```

Comment:
Without the `nullable = false` this gives us a pointer - but `Bytecode` aliases `[]byte` so we have the nil value if we want it, though ideally use `len(WASM) == 0` instead of `WASM == nil` if possible.

Reply:
The Bytecode type has some methods that assume EVM code is contained, for example the `Tokens` method tokenises the bytestring.

Suggest we have

- `type Bytecode []byte` for shared bytecode stuff (like marshalling)
- `type WASMBytecode Bytecode` for WASM
- `type EVMBytecode Bytecode` for EVM

In similar vein, I don't like having `Code` and `WASM`, I'd like to put them on an equal footing now have two code types. I suggest `EVMCode` and `WASMCode`.